### DISCUSSION TITLE
only MP3 outputs silent - file contains just LAME3.100UUUUUUUUUU????HUUUUUUUUUUUUUUUUUUUUUUUUUUUUUUUUUUUUUUUU repeated

### DISCUSSION THREAD

**Discussion by alecmyers** [rtl-airband/RTLSDR-Airband#398] [Q&A] [answered]
Summary: UDP output has correct audio, but all MP3 outputs are silent

version is 4.1.1. on RPi4B, compile flags are -DPLATFORM=armv8-generic

Frequency is set to local ATIS to produce continuous audio output.

Results:
- udp stream to local produces correct audio (which can be streamed to icecast with ffmpeg)
- direct icecast stream output is silence (stream connects correctly)
- file output contains repeated string LAME3.100UUUUUUUUUU????HUUUUUUUUUUUUUUUUUUUUUUUUUUUUUUUUUUUUUUUU...
i.e. looks like this: UUUUUUUUUUUUULAME3.100UUUUUUUUUU????HUUUUUUUUUUUUUUUUUUUUUUUUUUUUUUUUUUUUUUUULAME3.100UUUUUUUUUU????HUUUUUUUUUUUUUUUUUUUUUUUUUUUUUUUUUUUUUUUULAME3.100UUUUUUUUUU????HUUUUUUUUUUUUUUUUUUUUUUUUUUUUUUUUUUUUUUUULAME3.100UUUUUUUUUU????HUUUUUUUUUUUUUUUUUUUUUUUUUUUUUUUUUUUUUUUULAME3.100UUUUUUUUUU????HUUUUUUUUUUUUUUUUUUUUUUUUUUUUUUUUUUUUUUUULAME3.100UUUUUUUUUU????HUUUUUUUUUUUUUUUUUUUUUUUUUUUUUUUUUUUUUUUULAME3.100UUUUUUUUUU????HUUUUUUUUUUUUUUUUUUUUUUUUUUUUUUUUUUUUUUUULAME3.100UUUUUUUUUU????HUUUUUUUUUUUUUUUUUUUUUUUUUUU
- file plays back as silence

Seems like all MP3 coding is failing somewhere. Is this a problem with libmp3lame, or with RTLSDR-Airband? I'm not confident enough in the latter possibility to file a bug report. Any suggestions gratefully accepted.

dpkg -s libmp3lame-dev reports:

Package: libmp3lame-dev
Status: install ok installed
Priority: optional
Section: libdevel
Installed-Size: 732
Maintainer: Debian Multimedia Maintainers <debian-multimedia@lists.debian.org>
Architecture: arm64
Multi-Arch: same
Source: lame
Version: 3.100-3
Depends: libmp3lame0 (= 3.100-3)

The RTLSDR-airband config is as follows:

devices:
({
  type = "rtlsdr";
  index = 0;
  gain = 45;
  centerfreq = 133.55;
  correction = 0;
  sample_rate = 1.0;
  channels:  (
    {
      freq = 133.6;
      lowpass = 200;
      highpass = 4000; 
      squelch_threshold = -44;
      outputs: (
        {
	  type = "icecast";
	  server = "127.0.0.1";
          port = 8000;
          mountpoint = "xxx.mp3";
          name = "xxx";
          genre = "ATC";
          username = "xxx";
          password = "xxx";
	},
        {
          type="file";
          directory="/opt/xxx";
          filename_template="xxx";
          continuous=false;
          include_freq=false;
          append=true;
        },
        {
          type = "udp_stream";
          dest_address = "127.0.0.1";
          dest_port = 9001;
          continuous = true;
        }
      );
    }
  );
 }
);

**Comment by charlie-foxtrot** [comment, marked as answer]
I think your `highpass` / `lowpass` are flipped.  `highpass` will pass frequencies higher than what is specified, `lowpass` will pass frequencies lower than what is specified.  Not too many frequencies that are below 200Hz while at the same time above 4 kHz ;)

NOTE: `highpass` / `lowpass` only applies to the MP3 output and not to the UDP stream

**Comment by alecmyers** [comment]
D'oh! Thank you. Hours of my life spent chasing that one!

**Reply by charlie-foxtrot** [reply]
so others dont hit the same issue: #399
